hack/lint-go.sh

```diff
@@ -0,0 +1,6 @@
+if ! command -v golangci-lint &> /dev/null
+then
+    echo "WARNING golangci-lint could not be found, golang lint skipped.To install it: https://golangci-lint.run/usage/install/#local-installation"
```

Comment:
I would probably fail the script if we can't find it 
```suggestion
    echo "WARNING golangci-lint could not be found, golang lint skipped.To install it: https://golangci-lint.run/usage/install/#local-installation"
    exit 1
```

Reply:
+1 to failing the script as it more strongly encourages installing the linter

Reply:
Thank you both for the feedback 🙂
 Done!